### DISCUSSION TITLE
API to get top MITRE tactics of an agent

### DISCUSSION THREAD

**Discussion by ashutoshkumarmishra560** [wazuh/wazuh#18041] [Q&A]
Hi,

I am using Wazuh 4.4.5. I am trying to find the REST API to fetch the top MITRE tactic of a particular agent. I can see the details on the dashboard of Wazuh but unable to find the REST API to fetch the same.
Can you please point me to the right documentation for the same.

I can get the SCA violations but need the same for MITRE tactic for an agent.

Thanks

**Comment by MiguelazoDS** [comment]
Hi @ashutoshkumarmishra560, 

I think you're referring to this. https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.mitre_controller.get_tactics 

![image](https://github.com/wazuh/wazuh/assets/13010397/a4220714-5f79-48b6-abe1-e67935f59d4c)


Let us know if that helped. Regards!

**Reply by ashutoshkumarmishra560** [reply]
Thanks for taking out time and responding to the same. But I need the response for a particular agent. As shown in the screenshot
 when 
![ss](https://github.com/wazuh/wazuh/assets/11361005/7ab7f935-c581-4d86-bc9d-227cd518aaa1)

I view my agent in Wazuh Dashboard I get MITRE tactics related details such as Defense evasion, Privilege escalation etc but how to get the same data and details from REST API is something that I am struggling with.

**Reply by MiguelazoDS** [reply]
Mmm yeah, the information is already indexed it is not something you can get from a database that is the most common information using the API, the way I'm thinking you can get that information for all agents is using the CLI by making a request directly to the wazuh-indexer. 

Because as you can see, the dashboard information is just the filtered information 

![image](https://github.com/wazuh/wazuh/assets/13010397/d5f14205-4fdb-4066-9d4f-bb5b003f1591)

that you can get with these filters

![image](https://github.com/wazuh/wazuh/assets/13010397/c94dcf34-7de1-4ac3-8e7b-4d073d1afe44)

https://opensearch.org/docs/2.6/dashboards/run-queries/#working-in-the-curl-and-console-formats

**Comment by Dwordcito** [comment]
Closed for inactivity, please if you want to continue with the question, please reopen the discussion.

Thanks for being part of the community.

**Comment by winhigh** [comment]
@MiguelazoDS I have a question where i was using ossec previously but i use wazuh but i used to scrape ossec processed logs using scraping tools like prometheus but im unable to achieve the same using wazuh since wazuh  refuses to connect or needs auth to it - can you suggest any ideas to scrape logs from wazuh-ossec or mananger
